### PR TITLE
Fix Upgrade scenarios

### PR DESCRIPTION
**Pull request by shweta83**
### Problem Statement
Provisioning Template and Foreman Discovery Upgrade Scenario tests are failing with JSON parse error. 

### Solution

- Updated  `test_pre_scenario_provisioning_templates` to save hostname instead of host id in pre_upgrade_data file. 
- Saving FDI version in string format in `test_pre_upgrade_fdi_version`  as Version type data is not supported to be stored.

### CODE REVIEW

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/upgrades/test_discovery.py -k test_pre_upgrade_fdi_version -m pre_upgrade

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8788
Build Status: SUCCESS
PRT Comment: pytest tests/upgrades/test_discovery.py -k test_pre_upgrade_fdi_version -m pre_upgrade --external-logging
Test Result : =========== 1 passed, 1 deselected, 10 warnings in 898.68s (0:14:58) ===========
```

**Review by pondrejk — APPROVED** [review]

File: tests/upgrades/test_discovery.py [hunk 1]
```diff
@@ -60,5 +60,5 @@ def test_post_upgrade_fdi_version(self, target_sat, pre_upgrade_data):
         fdi_package = target_sat.execute('rpm -qa *foreman-discovery-image*').stdout
         # Note: The regular exp takes care of format digit.digit.digit or digit.digit.digit-digit in the output
         post_upgrade_version = Version(re.search(r'\d+\.\d+\.\d+(-\d+)?', fdi_package).group())
-        assert post_upgrade_version >= pre_upgrade_version
+        assert str(post_upgrade_version) >= pre_upgrade_version
```
Comment (Gauravtalreja1) [inline tests/upgrades/test_discovery.py:63]: comparing versions here make more sense here, so just would like to know why this change?

File: tests/upgrades/test_provisioningtemplate.py [hunk 2]
```diff
@@ -75,13 +73,11 @@ def test_pre_scenario_provisioning_templates(
 
         for kind in provisioning_template_kinds:
             assert host.read_template(data={'template_kind': kind})
-
         pre_update_data_dict = {
-            'provision_host_id': host.id,
+            'provision_host_name': host.name,
```
[the remark this answers is not in the document]
Comment (shweta83) [reply tests/upgrades/test_provisioningtemplate.py:77]: Host.name would be more reliable here. Id doesn't change

File: tests/upgrades/test_discovery.py (hunk 1 above)
Comment (shweta83) [reply tests/upgrades/test_discovery.py:63]: Since we dictionary doesn't accept Version so changing it to str.

**Review by pnovotny — APPROVED** [review]

File: tests/upgrades/test_discovery.py (hunk 1 above)
Comment (Gauravtalreja1) [reply tests/upgrades/test_discovery.py:63]: @shweta83 I didn't get, but as we discussed in our call we agreed to keep Version as is instead of comparing strings as its more effective

File: tests/upgrades/test_provisioningtemplate.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/upgrades/test_provisioningtemplate.py:77]: @shweta83 I think it doesn't matter if use either name or id, both should be same, but I'd like to understand rational behind changing it from id to name

File: tests/upgrades/test_discovery.py (hunk 1 above)
Comment (shweta83) [reply tests/upgrades/test_discovery.py]: Dictionary saved in pre_upgrade scenario doesn't accept Version object so I tried a new workaround for it.

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 8987
Build Status: SUCCESS
PRT Comment: pytest tests/upgrades/test_discovery.py -k test_pre_upgrade_fdi_version -m pre_upgrade --external-logging
Test Result : ========== 1 passed, 1 deselected, 10 warnings in 1808.27s (0:30:08) ===========
```

**Review by Gauravtalreja1 — APPROVED** [review]
